dotCMS/src/main/java/com/dotcms/contenttype/business/FieldAPIImpl.java

```diff
@@ -159,6 +159,10 @@ public void delete(Field field, User user) throws DotDataException, DotSecurityE
 
 		  perAPI.checkPermission(type, PermissionLevel.PUBLISH, user);
 
+		  Field oldField = fac.byId(field.id());
+		  if(oldField.fixed() || oldField.readOnly()){
```

Comment:
I just wonder if these methods should use the "is", such as isFixed, isReadOnly, isIndexed, etc